go/enclave/evm/transaction_overlay.go

```diff
@@ -0,0 +1,53 @@
+package evm
+
+import (
+	"math/big"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ten-protocol/go-ten/go/common"
+)
+
+func MaskedSender(address gethcommon.Address) gethcommon.Address {
+	return gethcommon.BigToAddress(big.NewInt(0).Sub(address.Big(), big.NewInt(1)))
+}
+
+func TransactionToMessageWithOverrides(
+	tx *common.L2PricedTransaction,
+	config *params.ChainConfig,
+	header *types.Header,
+) (*core.Message, error) {
+	// Override from can be used for calling system contracts from underivable addresses like all zeroes
+	if tx.FromSelf {
+		msg := TransactionToMessageNoSender(tx.Tx, header.BaseFee)
```

Comment:
why not just mutate the output of the gethCore stuff?
instead of creating the message by hand in the `TransactionToMessageNoSender` method?

Reply:
As I dont want to provide made up signatures that would error out in gethcore. Feels hackier than just recreating the whole thing